enterprise/backend/src/metabase_enterprise/workspaces/api.clj

```diff
@@ -134,22 +132,47 @@
                [:table :string]
                [:table_id [:maybe ::ws.t/appdb-id]]]]])
 
-(defn- batch-lookup-global-table-ids
-  "Batch lookup table_ids for global output tables by [db_id schema table]."
-  [db-id output-refs]
-  (when (seq output-refs)
-    (let [cases  (for [{:keys [schema table]} output-refs]
-                   [:and [:= :schema schema] [:= :name table]])
-          tables (t2/select :model/Table :db_id db-id {:where (into [:or] cases)})]
-      (into {} (map (juxt (juxt :schema :name) :id)) tables))))
-
+(defn- batch-lookup-global-table-ids*
+  "Given a bounded list of tables all, within the same database, return an association list of [db schema table] => id"
+  [db-id table-refs]
+  (t2/select-fn-vec (juxt (juxt (constantly db-id) :schema :name) :id)
+                    [:model/Table :id :schema [:name]]
+                    :db_id db-id
+                    {:where (into [:or] (for [{:keys [schema table]} table-refs]
+                                          [:and
+                                           [:= :schema schema]
+                                           [:= :name table]]))}))
+
+;; TODO
+(defn- batch-lookup-table-ids
+  "Given a list of maps holding [db_id schema table], return a mapping from those tuples => table_id"
+  [table-refs]
+  (when (seq table-refs)
+    ;; These are ordered by db, so this will partition fine.
+    (u/for-map [table-refs (partition-by :db_id table-refs)
+                :let [db_id (:db_id (first table-refs))]
+                ;; Guesstimating a number that prevents this query being too large.
+                table-refs (partition-all 20 table-refs)
```

Comment:
Maybe premature, but it really sucks when it's success has to ask for massive queries, and we see they're just truncated. I've been seen this too many times already.